### PR TITLE
Add loading bar

### PR DESCRIPTION
**Pull request by Joozty**
I am not very satisfied with the solution but it was the best I could find. Alternatively, we could have some sort of before and after hooks for API request but this would result in duplicate code because it needed to be done for every API request separately.

@viktor-yakubiv If you have a better solution I am all ears.

Closes https://github.com/oacore/dashboard/issues/188

### CODE REVIEW

File: store/root.js [hunk 1]
```diff
@@ -73,8 +89,8 @@ class Root {
     this.retrievePluginConfig()
 
     const url = `/data-providers/${this.dataProvider.id}`
-    this.works = new Works(url)
-    this.depositDates = new DepositDates(url)
+    this.works = new Works(url, this.apiRequest)
```
Comment (viktor-yakubiv) [inline store/root.js]: It's an options parameter, not a second initialisation parameter.

```suggestion
    this.works = new Works(url, this.options)
```

Despite of that, I propose to simply call it `request`

File: components/application/index.jsx [hunk 2]
```diff
@@ -21,6 +22,7 @@ const Application = ({
       <AppBar isAuthorized={isAuthorized}>
         {dataProvider && <RepositorySelect value={dataProvider} />}
         <Logout />
+        <LoadingBar />
```
Comment (viktor-yakubiv) [inline components/application/index.jsx]: Even if the loading bar is on top of application bar, it has nothing to the application bar.

**Comment by Joozty** [comment]
@viktor-yakubiv Go for it.

File: store/root.js (hunk 1 above)
Comment (Joozty) [reply store/root.js]: Maybe [passing `this`](https://mobx.js.org/best/store.html#combining-multiple-stores) would be even better. Obviously the serialization would be harder but we don't need it now anyway.

File: components/application/index.jsx (hunk 2 above)
Comment (Joozty) [reply components/application/index.jsx]: It's more about positioning and I liked it more having it below App bar.

**Comment by viktor-yakubiv** [comment]
I like the loading bar more in the top. Maybe because it's more common. It's personal preference anyway. We can ask the team if you wish.

The goal is to make the only root component control everything and avoid side effects like `withGlobalStore` where it's possible.

To achieve the goal I had to make global `App` being `observer` @Joozty please let me know if you see significant performance issues.

If you think it's fine, feel free to rebase and merge.

**Comment by Joozty** [comment]
> I like the loading bar more in the top. Maybe because it's more common. It's personal preference anyway. We can ask the team if you wish.

That's fine it looks even better now. 👍 

> The goal is to make the only root component control everything and avoid side effects like withGlobalStore where it's possible.

I just want to point out that using `withGlobalStore` and observer is basically the same. `withGlobalStore` adds just `store` prop on top of that but you know that already I guess. It just cannot be used on app component.


> To achieve the goal I had to make global App being observer @Joozty please let me know if you see significant performance issues.

I personally consider a very bad practice using `@observe` on root component. 

Let me explain. The application is rerendered 5 times more.
Every request triggers rerender - 4  request
One rerender is called when loading bar needs to be hidden

Like it's not a big deal because react is really good with virtual DOM but I still think only Loading bar should be observer.

And you removed my best feature - letting animation to finish before hiding it. Why? I was thinking it was quite cool. 😄

**Comment by viktor-yakubiv** [comment]
Yes, this is what I really sorry for. On the other hand, I am not sure it's correct if animation goes when the data is fetched.

**Comment by Joozty** [comment]
@viktor-yakubiv Sometimes I do things that are not even necessary 😄

**Comment by Joozty** [comment]
What's the conclusion of this one? Are you going to change it or you want to leave it as is?

**Comment by viktor-yakubiv** [comment]
I would leave this as is but I needed to do leftover here.

**Comment by viktor-yakubiv** [comment]
I checked the code, I would leave this as is unless we have performance issues

**Pull request merged**